packages/url/src/is-url.native.js

```diff
@@ -0,0 +1,18 @@
+const URL_REGEXP = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[\-;:&=\+\$,\w]+@)?[A-Za-z0-9\.\-]+|(?:www\.|[\-;:&=\+\$,\w]+@)[A-Za-z0-9\.\-]+)(?::\d{2,5})?((?:\/[\+~%\/\.\w\-_]*)?\??(?:[\-\+=&;%@\.\w_]*)#?(?:[\.\!\/\\\w]*))?)/i;
+
+/**
+ * Determines whether the given string looks like a URL.
+ *
+ * @param {string} url The string to scrutinise.
+ *
+ * @example
+ * ```js
+ * const isURL = isURL( 'https://wordpress.org' ); // true
+ * ```
+ *
+ * @return {boolean} Whether or not it looks like a URL.
+ */
+export function isURL( url ) {
+	const match = url.match( URL_REGEXP );
+	return match !== null && match.length >= 1 && match[ 0 ] === url;
```

Comment:
It was a bit confusing for me to understand why it wasn't enough to implement this as simply:

```js
return URL_REGEXP.test( url );
```

Running it locally, I can see it doesn't work, and it seems like the pattern matches a lot of "invalid" URLs that must be subjected to further conditions. It would help if there were an inline code comment explaining the logic of what is happening here.

Reply:
Added on https://github.com/WordPress/gutenberg/pull/20172/commits/c16b34e959914e2bacc3e0daa1db52f0a5641104 👍